### DISCUSSION TITLE
How to assume all Jupyter Book's MyST files with code cells are jupytext without frontmatter

### DISCUSSION THREAD

**Discussion by ketozhang** [executablebooks/meta#1204] [Q&A]
I have many notes that have executable content. I do not want to this long frontmatter to all of them:

```markdown
---
jupytext:
  formats: md:myst
  text_representation:
    extension: .md
    format_name: myst
kernelspec:
  display_name: Python 3
  language: python
  name: python3
---
```

Is there a shorter way to mark executable documents or a config to assume all markdown files are executable MyST markdown?

**Comment by agoose77** [comment]
Hi @ketozhang,

My answer here is not the definitive answer; it's a personal opinion that I hope reflects the wider ethos of our collaboration. 

In MyST / JB, we're trying to capture as much information about reproducibility as possible. I think it's important that a floating Markdown file can at least contain enough information about _how_ to execute it, even if that's just a string that requires additional context in `myst.yml` or `_config.yml` to interpret.

The important information is not so much "this is executable", but rather "this file uses <XXX> kernel". In Jupyter Book 2 / MyST-MD, we're thinking about wider execution improvements such as execution _strategies_ that allow mixing local and remote execution. We could look to lift the kernel frontmatter up to the project, so that any file with code cells / eval expressions would use this metadata. At this point, Jupyter Book 1 is in its maintenance phase, so we probably won't architect such a new feature there. 

Does this answer your question? :)

**Reply by chrisjsewell** [reply]
> The important information is not so much "this is executable", but rather "this file uses kernel"

👍

**Comment by chrisjsewell** [comment]
> Is there a shorter way to mark executable documents

yes:

https://myst-nb.readthedocs.io/en/latest/authoring/basics.html#text-based-notebooks

```
---
file_format: mystnb
kernelspec:
  name: python3
---
```
